OpCode::EditorOpen: return 1 if editor opened successfully

crsaracco opened the pull request:
`dispatch()` should return 1 if the editor was opened successfully. Some DAWs ignore this return value, but others don't.

One example of a DAW that doesn't is Carla ([see this chunk of code](https://github.com/falkTX/Carla/blob/51f20736b4e0b5d6416f2b7d4051f155f5858fbf/source/backend/plugin/CarlaPluginVST2.cpp#L513)).

Since we return 0 here by default, Carla assumes that the editor failed to open.

I think the more correct way to do this would be to actually check that `editor.open()` was successful in some way, but that would require an API change.

askeksa commented:
I would suggest that we change the editor API so `editor.open()` returns a `bool` indicating success. This seems to be a method where it is relevant to report failure (even though some DAWs ignore it).

crsaracco commented:
Updated: now `Editor::open()` returns a bool, as suggested.

I also updated the documentation comment for `Editor::open()` regarding how the parent pointer should be interpreted on different platforms -- requesting feedback from OS X (@robsaunders) and Windows (?) users to make sure the comment is correct.

askeksa commented on src/editor.rs:
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
+    ///  - On Mac OS X (64 bit), it should be interpreted as a `NSView*`
+    ///  - On X11 platforms, it should be interpreted as a `u32` (the ID number of the parent window)
+    ///
+    /// Return true if the window opened successfully, false otherwise.
```
Put `true` and `false` in backquotes.

crsaracco replied:
Thanks! Changed all of these in the `Editor` trait comments.

askeksa commented on src/editor.rs:
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
-
     /// Set the knob mode for this editor (if supported by host).
     ///
     /// Return true if the knob mode was set.
```
Put `true` in backquotes here as well, while you are at it fixing typos and such. :)

askeksa commented on src/editor.rs:
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
     }
 
-    /// Recieve key up event. Return true if the key was used.
+    /// Receive key up event. Return true if the key was used.
```
Put `true` in backquotes. Also below in `key_down`.

Walther commented:
Tested on
- MacOS Mojave 10.14.1 (18B75)
- https://github.com/crsaracco/rtb-rs-tester at commit `948a3e7`
- Bitwig Studio 2.4.2

The Rutabaga window loads up 🎉 
<img width="863" alt="screen shot 2019-01-19 at 17 36 27" src="https://user-images.githubusercontent.com/2943750/51434259-82715d00-1c11-11e9-9898-8e18b1db7c59.png">
Clicking inside it crashes the plugin 😢 
<img width="282" alt="screen shot 2019-01-19 at 17 36 43" src="https://user-images.githubusercontent.com/2943750/51434260-82715d00-1c11-11e9-828a-dcb1b800b46d.png">
Reloading the plugin works, but requires additional press of the "Show window" toggle. Toggling that indeed shows and hides the plugin window.
<img width="573" alt="screen shot 2019-01-19 at 17 37 05" src="https://user-images.githubusercontent.com/2943750/51434261-82715d00-1c11-11e9-86c9-3966e739e01b.png">

Then again I was just told in chat that Bitwig ignores the return value, so 🤷‍♂️

askeksa commented:
Tested on Windows 10 via `rtb-rs-tester` in Renoise, Carla and Bitwig Studio. Window is opening and closing fine in all situations.

askeksa commented:
What is keeping this from being merged?
